### PR TITLE
Add first edge probing task example (SPR1)

### PR DESCRIPTION
**Pull request by pyeres**
This PR adds a first edge probing task example: SPR1. These changes contain an implementation of SPR1, but most of the code (~300 lines) goes toward implementing components that will be shared by the other edge probing tasks. 

**Important notes**:
* Commits [7808e53,... 96b18a1] deal with the token aligner and are up for review in PR #80 (i.e., **`retokenize.py`, `test_token_alignment.py`, and `requirements.txt` can be ignored in this review**).
* All the rest of the commits [94260e0, ...bf0de62] are toward implementing two-span, multi-label edge probing tasks (specifically SPR1 here). These changes need review.

### CODE REVIEW

File: jiant/tasks/lib/templates/edge_probing.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        target_span2 = aligner.project_span(self.span2[0], self.span2[1])
+        return TokenizedExample(
+            guid=self.guid,
+            tokens=target_tokenization,
+            span1_span=target_span1,
+            span2_span=target_span2,
+            span1_text=" ".join(target_tokenization[target_span1[0] : target_span1[1]]),
+            span2_text=" ".join(target_tokenization[target_span2[0] : target_span2[1]]),
+            label_ids=[self.task.LABEL_TO_ID[label] for label in self.labels],
+            label_num=len(self.task.LABELS),
```
Comment (pyeres) [inline jiant/tasks/lib/templates/edge_probing.py:57]: @zphang — is there a better way for me to get access to the number of labels for use in the featurization stage (than passing them through here)?

File: jiant/tasks/evaluate/core.py [hunk 2]
```diff
@@ -111,6 +111,30 @@ def compute_metrics_from_preds_and_labels(cls, preds, labels):
...
+        return (logits > 0.5).astype(int)
+
+    @classmethod
+    def compute_metrics_from_preds_and_labels(cls, preds, labels):
+        # noinspection PyUnresolvedReferences
+        acc = float((preds == labels).mean())
+        labels = np.array(labels)
+        minor = {
+            "acc": acc,
+            "f1_micro": f1_score(y_true=labels, y_pred=preds, average="micro"),
```
Comment (pyeres) [inline jiant/tasks/evaluate/core.py:129]: There was some confusion recently about which F1 measure was appropriate. I threw them all in for testing. These will be trimmed before merging this PR.

File: jiant/tasks/retrieval.py [hunk 3]
```diff
@@ -58,6 +59,7 @@
     "scitail": SciTailTask,
     "snli": SnliTask,
     "socialiqa": SocialIQATask,
+    "spr1": Spr1Task,
```
Comment (zphang) [inline jiant/tasks/retrieval.py:62]: One thing to consider (and I don't have a settled stance on this) is whether each task should be its own implementation. E.g. Both SQuAD v1 and v2 have a single implementation, for cross-lingual transfer tasks (not checked in yet) all language-versions of each task has a single implementation, etc. Can discuss.

File: jiant/tasks/evaluate/core.py (hunk 2 above)
Comment (zphang) [reply jiant/tasks/evaluate/core.py:129]: Are these still being trimmed?

Comment (pyeres) [reply jiant/tasks/evaluate/core.py:129]: Thanks for the reminder, trimmed w/ fb5f135.
